exec/java-exec/src/main/resources/rest/index.ftl

```diff
@@ -179,6 +184,98 @@
         </div>
       </div>
   </div>
+   <script charset="utf-8">
+      var refreshTime = 2000;
+      var refresh = getRefreshTime();
+      var timeout;
+      var size = $("#size").html();
+
+      function getRefreshTime() {
+          var refresh = $.ajax({
+                          type: 'GET',
+                          url: '/graceperiod',
+                          dataType: "json",
+                          complete: function(data) {
+                                refreshTime = data.responseJSON["graceperiod"];
+                                refreshTime = refreshTime/3;
+                                timeout = setTimeout(reloadStatus,refreshTime );
+                                }
+                          });
+      }
+      function reloadStatus () {
+          console.log(refreshTime);
+          var result = $.ajax({
+                      type: 'GET',
+                      url: '/state',
+                      dataType: "json",
+                      complete: function(data) {
+                            fillStatus(data,size);
+                            }
+                      });
+          timeout = setTimeout(reloadStatus, refreshTime);
+      }
+
+      function fillStatus(data,size) {
+          var status_map = (data.responseJSON);
+          for (i = 1; i <= size; i++) {
+            var address = $("#row-"+i).find("#address").contents().get(0).nodeValue;
+            address = address.trim();
+            var port = $("#row-"+i).find("#port").html();
+            var key = address+"-"+port;
+
+            if (status_map[key] == null) {
+                $("#row-"+i).find("#status").text("OFFLINE");
+                $("#row-"+i).find("#shutdown").prop('disabled',true).css('opacity',0.5);
+                $("#row-"+i).find("#queriesCount").text("");
+            }
+            else {
+                if( status_map[key] == "ONLINE") {
+                    $("#row-"+i).find("#status").text(status_map[key]);
+                }
+                else {
+                    fillQueryCount(address,i);
+                    $("#row-"+i).find("#status").text(status_map[key]);
+                }
+            }
+          }
+      }
+      function fillQueryCount(address,row_id) {
+          port_num = 8047
```

Comment:
Port number is not always 8047. It can be configured via `drill.exec.http.port`.